cmd/geth/chaincmd.go

```diff
@@ -206,6 +221,64 @@ func getIsQuorum(file io.Reader) bool {
 	return altGenesis.Config.IsQuorum == nil || *altGenesis.Config.IsQuorum
 }
 
+// updateTransitions will update genesis block with the new transitions data
+func updateTransitions(ctx *cli.Context) error {
+	// Open and initialise both full and light databases
+	stack, _ := makeConfigNode(ctx)
+	defer stack.Close()
+
+	// Make sure we have a valid genesis JSON
+	genesisPath := ctx.Args().First()
+	if len(genesisPath) == 0 {
+		utils.Fatalf("Must supply path to genesis JSON file")
+	}
+	file, err := os.Open(genesisPath)
+	if err != nil {
+		utils.Fatalf("Failed to read genesis file: %v", err)
+	}
+	defer file.Close()
+
+	genesis := new(core.Genesis)
+	if err := json.NewDecoder(file).Decode(genesis); err != nil {
+		utils.Fatalf("invalid genesis file: %v", err)
+	}
+
+	// Quorum
+	file.Seek(0, 0)
+	genesis.Config.IsQuorum = getIsQuorum(file)
+
+	if genesis.Config.IsQuorum {
+		err = genesis.Config.CheckTransitionsData()
+		if err != nil {
+			utils.Fatalf("transitions data invalid: %v", err)
+		}
+	} else {
+		return fmt.Errorf("update transitions only apply to quorum configuration")
+	}
+
+	// Update transitions and recommit to db
+	for _, name := range []string{"chaindata", "lightchaindata"} {
+		chaindb, err := stack.OpenDatabase(name, 0, 0, "", false)
+		if err != nil {
+			utils.Fatalf("Failed to open database: %v", err)
+		}
+		stored := rawdb.ReadCanonicalHash(chaindb, 0)
+		storedcfg := rawdb.ReadChainConfig(chaindb, stored)
+		if storedcfg == nil {
+			return fmt.Errorf("found genesis block without chain config")
+		}
+		// Check that new transitions have changed before updating them
+		if !reflect.DeepEqual(storedcfg.Transitions, genesis.Config.Transitions) {
+			log.Info("Change found in transitions, proceeding to update chain config")
+			storedcfg.Transitions = genesis.Config.Transitions
+			rawdb.WriteChainConfig(chaindb, stored, storedcfg)
+		} else {
+			log.Info("No change in transitions, no update required to chain config")
+		}
+	}
+	return nil
```

Comment:
Perhaps we should return 'true' if a change was made, and false if nothing. If this goes back to the api caller.

Reply:
not modifying as this is a nice-to-have that might disrupt geth api wrapping - to investigate separately.

Reply:
rest of the chainCmd and accountCmd only returns an error, suggest we keep it consistent for now. Making a change to the interface requires an entire round of regression testing, will revisit this in future